### PR TITLE
[Build] Support excluding C++ examples from parent builds

### PR DESCRIPTION
**Pull request by rokups**
Allow to exclude examples from build by setting ATOMIC_CPLUSPLUS_EXAMPLES in parent project.

Export ATOMIC_CXX11, ATOMIC_64BIT, ATOMIC_BUILD_DIST (allows parent projects to use Atomic's c++11 support).

### CODE REVIEW

File: Build/CMake/Modules/AtomicCommon.cmake [hunk 1]
```diff
@@ -22,19 +22,49 @@
 
 include (CMakeParseArguments)
 
+# Followint three `*_exported()` macros are used defining global cmake variables with game engine include directories,
```
Comment (JoshEngebretson) [inline Build/CMake/Modules/AtomicCommon.cmake]: Tiny little typo on "Followint" didn't catch before

File: CMakeLists.txt [hunk 2]
```diff
@@ -78,7 +78,9 @@ add_definitions(${ATOMIC_DEFINITIONS})
 add_subdirectory(Source)
 
 if (ATOMIC_DESKTOP AND ATOMIC_DEV_BUILD)
-    set (ATOMIC_CPLUSPLUS_EXAMPLES 1)
+    if (NOT DEFINED ATOMIC_CPLUSPLUS_EXAMPLES)
+        set (ATOMIC_CPLUSPLUS_EXAMPLES 1)
+    endif ()
```
Comment (JoshEngebretson) [inline CMakeLists.txt]: Good add

File: Build/CMake/Modules/AtomicCommon.cmake (hunk 1 above)
Comment (rokups) [inline Build/CMake/Modules/AtomicCommon.cmake]: Amended commit that moved code fixing typo

**Comment by JoshEngebretson** [comment]
Great, thanks, LANDING

**Pull request merged**